src/views/author_notes.ts

```diff
@@ -14,14 +14,114 @@
+  allowedTypes = ['corresp', 'footnote', 'paragraph']
+  headersStrings: { [key: string]: string } = {
+    corresp: 'Correspondence',
+    footnote: 'Author notes',
+    paragraph: 'Author notes',
+  }
 
   public ignoreMutation = () => true
   public stopEvent = () => true
+
+  public initialise = () => {
+    this.createDOM()
+    this.createGutter('block-gutter', this.gutterButtons().filter(Boolean))
+    this.createElement()
+    this.createGutter(
+      'action-gutter',
+      this.actionGutterButtons().filter(Boolean)
+    )
+    this.updateContents()
+  }
+
+  public createElement = () => {
+    this.container = document.createElement('div')
+    this.container.classList.add('block', 'author-notes-container')
+    this.dom.appendChild(this.container)
+  }
+
+  public createDOM = () => {
+    this.dom = document.createElement('div')
+    this.dom.classList.add('block-container')
+    this.dom.classList.add(`block-${this.node.type.name}`)
+    this.dom.setAttribute('contenteditable', 'false')
+  }
+
+  public updateContents = () => {
+    this.container.innerHTML = ''
+
+    const items: { [key: string]: ManuscriptNode[] } = {}
+
+    this.view.state.doc.descendants((node, pos) => {
+      if (node.type.name === 'author_notes') {
+        node.descendants((anNode, anPos) => {
+          items[anNode.type.name] = []
+          if (this.allowedTypes.includes(anNode.type.name)) {
+            items[anNode.type.name].push(anNode)
+            return false
+          }
+        })
+        return false
+      }
+    })
+
+    this.allowedTypes.forEach((type) => {
+      if (items[type] && items[type].length) {
+        const showHeader =
+          type === this.allowedTypes[0] ||
+          (type === this.allowedTypes[1] &&
+            (!!items[this.allowedTypes[1]]?.length ||
+              !!items[this.allowedTypes[2]]?.length)) ||
+          (type === this.allowedTypes[2] &&
+            !items[this.allowedTypes[1]]?.length &&
+            !!items[this.allowedTypes[2]]?.length)
+
+        const content = this.createContent(type, items[type], showHeader)
+        if (content) {
+          this.container.appendChild(content)
+        }
+      }
+    })
+  }
+
+  public createHeader = (type: string) => {
+    const header = document.createElement('h1')
+    header.classList.add(`${type}-title`, 'block')
+    header.textContent = this.headersStrings[type]
+    return header
+  }
+
+  public createContent = (type: string, nodes: ManuscriptNode[], showHeader: boolean) => {
+    let isNotEmpty = false
+    if (nodes.length) {
+      const container = document.createElement('div')
+      container.classList.add(`${type}s-container`)
+
+      if (showHeader) {
+        container.append(this.createHeader(type))
+      }
+      nodes.forEach((node) => {
+        if (node.textContent) {
+          isNotEmpty = true
+          const div = document.createElement('div')
+          div.classList.add(type)
+          div.append(node.textContent)
```

Comment:
please use textContent instead of append

Reply:
OK...